pkg/controller/release/release_controller.go

```diff
@@ -0,0 +1,659 @@
+
+		installationTargetLister:  installationTargetInformer.Lister(),
+		installationTargetsSynced: installationTargetInformer.Informer().HasSynced,
+
+		trafficTargetLister:  trafficTargetInformer.Lister(),
+		trafficTargetsSynced: trafficTargetInformer.Informer().HasSynced,
+
+		capacityTargetLister:  capacityTargetInformer.Lister(),
+		capacityTargetsSynced: capacityTargetInformer.Informer().HasSynced,
+
+		releaseWorkqueue: workqueue.NewNamedRateLimitingQueue(
+			workqueue.DefaultControllerRateLimiter(),
+			"release_controller_releases",
+		),
+		applicationWorkqueue: workqueue.NewNamedRateLimitingQueue(
+			workqueue.DefaultControllerRateLimiter(),
+			"release_controller_applications",
+		),
+	}
+
+	glog.Info("Setting up event handlers")
+
+	releaseInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: controller.enqueueRelease,
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				controller.enqueueRelease(newObj)
+			},
+		})
+
+	installationTargetInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: controller.enqueueInstallationTarget,
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				controller.enqueueInstallationTarget(newObj)
+			},
+		})
+
+	capacityTargetInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: controller.enqueueCapacityTarget,
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				controller.enqueueCapacityTarget(newObj)
+			},
+		})
+
+	trafficTargetInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: controller.enqueueTrafficTarget,
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				controller.enqueueTrafficTarget(newObj)
+			},
+		})
+
+	return controller
+}
+
+func (c *Controller) Run(threadiness int, stopCh <-chan struct{}) {
+	defer runtime.HandleCrash()
+	defer c.releaseWorkqueue.ShutDown()
+	defer c.applicationWorkqueue.ShutDown()
+
+	glog.V(2).Info("Starting Release controller")
+	defer glog.V(2).Info("Shutting down Release controller")
+
+	if ok := cache.WaitForCacheSync(
+		stopCh,
+		c.applicationsSynced,
+		c.releasesSynced,
+		c.clustersSynced,
+		c.installationTargetsSynced,
+		c.trafficTargetsSynced,
+		c.capacityTargetsSynced,
+	); !ok {
+		runtime.HandleError(fmt.Errorf("failed to wait for caches to sync"))
+		return
+	}
+
+	for i := 0; i < threadiness; i++ {
+		go wait.Until(c.runReleaseWorker, time.Second, stopCh)
+		go wait.Until(c.runApplicationWorker, time.Second, stopCh)
+	}
+
+	glog.V(4).Info("Started Release controller")
+
+	<-stopCh
+}
+
+func (c *Controller) runReleaseWorker() {
+	for c.processNextReleaseWorkItem() {
+	}
+}
+
+func (c *Controller) runApplicationWorker() {
+	for c.processNextAppWorkItem() {
+	}
+}
+
+func (c *Controller) processNextReleaseWorkItem() bool {
+	obj, shutdown := c.releaseWorkqueue.Get()
+	if shutdown {
+		return false
+	}
+
+	defer c.releaseWorkqueue.Done(obj)
+
+	if _, ok := obj.(string); !ok {
```

Comment:
```suggestion
	if key, ok := obj.(string); !ok {
```
The assignment to `key` a few lines below could be then removed.